python/ray/data/read_api.py

```diff
@@ -711,3 +718,9 @@ def _get_metadata(table: "pyarrow.Table") -> BlockMetadata:
     stats = BlockExecStats.builder()
     return BlockAccessor.for_block(table).get_metadata(
         input_files=None, exec_stats=stats.build())
+
+
+def _prepare_read(ds: Datasource, ctx: DatasetContext, parallelism: int,
+                  **kwargs) -> List[ReadTask]:
+    DatasetContext._set_current(ctx)
+    return ds.prepare_read(parallelism, **kwargs)
```

Comment:
Pretty sure that this will break when `**kwargs` contains an Arrow `S3FiileSystem` due to those outstanding serialization issues, which is why we currently have this serialization wrapper for it before serializing the filesystem for the task. Might need to use that serialization wrapper before `prepare_read.remote()`.

Reply:
Good catch.